x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/persistence/InferenceIndexConstants.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ml.inference.persistence;
+
+/**
+ * Class containing the index constants so that the index version, name, and prefix are available to a wider audience.
+ */
+public final class InferenceIndexConstants {
+
+    public static final String INDEX_VERSION = "000001";
+    public static final String INDEX_NAME_PREFIX = ".ml-inference-";
+    public static final String INDEX_PATTERN = INDEX_NAME_PREFIX + "*";
+    public static final String LATEST_INDEX_VERSIONED_NAME = INDEX_NAME_PREFIX + INDEX_VERSION;
+    public static final String LATEST_INDEX_NAME = LATEST_INDEX_VERSIONED_NAME;
```

Comment:
Are we going to use non-versioned index name in some circumstances? If so, when?

Reply:
What do you mean "non-versioned index name"? 

The only two uses that i can think of are querying `.ml-inference-*` for all indices or calling indexing against `.ml-inference-LATEST`. 

I can get rid of `LATEST_INDEX_VERSIONED_NAME` for simplicity.